clang/lib/Sema/SemaDecl.cpp

```diff
@@ -19240,6 +19240,25 @@ void Sema::ActOnFields(Scope *S, SourceLocation RecLoc, Decl *EnclosingDecl,
 
     if (Record && FD->getType().isVolatileQualified())
       Record->setHasVolatileMember(true);
+    auto IsNonDependentBitField = [](const FieldDecl *FD) {
+      return FD->isBitField() && !FD->getType()->isDependentType();
+    };
+
+    if (Record && PreviousField && IsNonDependentBitField(FD) &&
```

Comment:
If this is not enabled by default, maybe we should check the diagnostic is not ignored to avoid unnecessary work (`DiagnosticsEngine::isIgnored`)